src/l1-resources/data-lake-pipeline/index.ts

```diff
@@ -0,0 +1,374 @@
+    (r, i) => (i[1] === undefined ? r : { ...r, [i[0]]: i[1] }),
+    {}
+  );
+}
+/* eslint-enable max-len, quote-props */
+
+/**
+ * Type of ingestion destination of this Data Lake Pipeline.
+ *
+ * @schema SinkType
+ */
+export enum SinkType {
+  /** DLS */
+  DLS = "DLS",
+}
+
+/**
+ * Ordered fields used to physically organize data in the destination.
+ *
+ * @schema partitionFields
+ */
+export interface PartitionFields {
+  /**
+   * Human-readable label that identifies the field name used to partition data.
+   *
+   * @schema partitionFields#FieldName
+   */
+  readonly fieldName?: string;
+
+  /**
+   * Sequence in which MongoDB Cloud slices the collection data to create partitions. The resource expresses this sequence starting with zero.
+   *
+   * @schema partitionFields#Order
+   */
+  readonly order?: number;
+}
+
+/**
+ * Converts an object of type 'PartitionFields' to JSON representation.
+ */
+/* eslint-disable max-len, quote-props */
+export function toJson_PartitionFields(
+  obj: PartitionFields | undefined
+): Record<string, any> | undefined {
+  if (obj === undefined) {
+    return undefined;
+  }
+  const result = {
+    FieldName: obj.fieldName,
+    Order: obj.order,
+  };
+  // filter undefined values
+  return Object.entries(result).reduce(
+    (r, i) => (i[1] === undefined ? r : { ...r, [i[0]]: i[1] }),
+    {}
+  );
+}
+/* eslint-enable max-len, quote-props */
+
+/**
+ * Type of ingestion source of this Data Lake Pipeline.
+ *
+ * @schema SourceType
+ */
+export enum SourceType {
+  /** ON_DEMAND_CPS */
+  ON_DEMAND_CPS = "ON_DEMAND_CPS",
+  /** PERIODIC_CPS */
+  PERIODIC_CPS = "PERIODIC_CPS",
+}
+
+/**
+ * A CloudFormation `MongoDB::Atlas::DataLakePipeline`
+ *
+ * @cloudformationResource MongoDB::Atlas::DataLakePipeline
+ * @stability external
+ */
+export class CfnDataLakePipeline extends cdk.CfnResource {
+  /**
+   * The CloudFormation resource type name for this resource class.
+   */
+  public static readonly CFN_RESOURCE_TYPE_NAME =
+    "MongoDB::Atlas::DataLakePipeline";
+
+  /**
+   * Resource props.
+   */
+  public readonly props: CfnDataLakePipelineProps;
+
+  /**
+   * Attribute `MongoDB::Atlas::DataLakePipeline.CreatedDate`
+   */
+  public readonly attrCreatedDate: string;
+  /**
+   * Attribute `MongoDB::Atlas::DataLakePipeline.Id`
+   */
+  public readonly attrId: string;
+  /**
+   * Attribute `MongoDB::Atlas::DataLakePipeline.LastUpdatedDate`
+   */
+  public readonly attrLastUpdatedDate: string;
+
+  /**
+   * Create a new `MongoDB::Atlas::DataLakePipeline`.
+   *
+   * @param scope - scope in which this resource is defined
+   * @param id    - scoped id of the resource
+   * @param props - resource properties
+   */
+  constructor(
+    scope: constructs.Construct,
+    id: string,
+    props: CfnDataLakePipelineProps
+  ) {
+    super(scope, id, {
+      type: CfnDataLakePipeline.CFN_RESOURCE_TYPE_NAME,
+      properties: toJson_CfnDataLakePipelineProps(props)!,
```

Comment:
would you please show me where are we testing that all these nested `toJson_...` methods are covered with unit tests?

Reply:
Hi, This method will be called internally while constructor is created. In file "integ.default.ts" line 45 will call this internally.